### ISSUE TITLE
__name__ being lost inside cloudpickled staticmethods? (0.8.0 regression)

### ISSUE THREAD

**Issue by tjb900** [cloudpipe/cloudpickle#251]
Not sure if this is expected or not?  Reproducer below.

Thanks!
Tim

----

Output with cloudpickle 0.7.0:

```
(cp7) $ python reproducer.py                                                                                                                                   
<pkg.mymod2.MyClass object at 0x7fbf18b3d828>
```

cloudpickle 0.8.0:

```python-traceback
(cp8) $ python reproducer.py                                                                                                                                   
Traceback (most recent call last):
  File "reproducer.py", line 7, in <module>
    print(f())
  File "/mypath/pkg/mymod.py", line 4, in func
    from .mymod2 import MyClass
KeyError: "'__name__' not in globals"
```

```
pkg/
   __init__.py
   mymod.py
   mymod2.py
reproducer.py
```

```python
# pkg/__init__.py
# blank
```

```python
# pkg/mymod.py
class Blah(object):
    @staticmethod
    def func():
        from .mymod2 import MyClass
        return MyClass()
```

```python
# pkg/mymod2.py
class MyClass(object):
    pass
```

```python
# reproducer.py
from pkg.mymod import Blah
from cloudpickle import dumps
from pickle import loads

f = loads(dumps(Blah.func))

print(f())
```

**Comment by pierreglaser** [comment]
You're correct, its a bug.
In `0.8.0` we reverted functions `globals` handling to the behavior of `0.5.3`, and accidentally reintroduced a bug of `0.5.3` that was never reported before. 
It is not related to `staticmethod` specifically, it is due to the relative import mechanism (the failing line is 
`from .mymod2 import MyClass`)
 I'll do a PR to fix this.

**Comment by ogrisel** [comment]
I released 0.8.1 to PyPI this morning to publish this fix.
